user-guide/src/overview.md

```diff
@@ -0,0 +1,24 @@
+# Overview
+
+In short, when writing a daemon or a service, we have the *"muscle"* of the
+application ‒ whatever we write the daemon for. And we have a whole lot of
+infrastructure around that: logging, command line parsing, configuration. And
+while there are Rust libraries for all that, one needs nontrivial amount of
+boilerplate code to bridge all this together.
+
+Spirit aims to be this bridge. 
+
+It takes care of things like
+
+- signal handling and the application lifecycle
+- combining multiple pieces of configuration together with command line
+  overrides
+- it allows for reloading the configuration at runtime
+- application metrics
```

Comment:
Style nitpick/question: should the bullet points have a dot at their end? I don't know about any official rules and I'm not native English speaker, but to me this way they feel a little bit „naked“ without anything.